### DISCUSSION TITLE
How to disable automatic decompression?

### DISCUSSION THREAD

**Discussion by william-silversmith** [encode/httpx#1794] [Q&A] [answered]
Hi all,

I've been experimenting with http2 using async and comparing it with my preexisting http1.1 threaded implementation. I use my own choice of libraries and handling of decompression and would like to disable it in httpx while retaining a response object. What would be the right way to go about that?

Thanks for your help!

PS: For reference, I'm experimenting with seeing how much additional performance I can squeeze out of https://github.com/seung-lab/cloud-files

**Comment by lovelydinosaur** [comment, marked as answer]
You can use `.iter_raw()` to iterate over the raw uncompressed byte content.

**Reply by william-silversmith** [reply]
Thank you Tom! I wrote this which seems to work:

```python
    async def httpx_get_raw(client, path, headers):
      raw = b''
      async with client.stream("GET", path, headers=headers) as resp:
        async for raw_bytes in resp.aiter_raw():
          raw += raw_bytes
      return resp, raw
```
